source/common/http/conn_manager_impl.cc

```diff
@@ -708,12 +708,19 @@ void ConnectionManagerImpl::ActiveStream::decodeHeaders(HeaderMapPtr&& headers,
     stream_info_.setDownstreamRemoteAddress(ConnectionManagerUtility::mutateRequestHeaders(
         *request_headers_, connection_manager_.read_callbacks_->connection(),
         connection_manager_.config_, *snapped_route_config_, connection_manager_.random_generator_,
-        connection_manager_.runtime_, connection_manager_.local_info_));
+        connection_manager_.local_info_));
   }
   ASSERT(stream_info_.downstreamRemoteAddress() != nullptr);
 
   ASSERT(!cached_route_);
   refreshCachedRoute();
+
+  if (!state_.is_internally_created_) {
```

Comment:
Sorry why is this new check necessary? Can you add a comment?

Reply:
Sure can add a comment - essentially previously the call to `mutateTracingRequestHeader` was performed inside `mutateRequestHeaders` which was subject to the same condition just above. So thought, even though the call has been pulled out from `mutateRequestHeaders`, it should still be subject to the same condition? Let me know if not correct.

Is this condition only true for the initial processing of a request at the first proxy? In which case this would prevent the request id being modified mid way through a mesh I assume.